### PR TITLE
Gives effects to drinks Alexander, Between the Sheets, and Menthol

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
:cl: Cobby
tweak: Several of the recent drinks added have now been given effects!
add: Alexander: Conquer the world... or just the station with this drink! If you have a shield, it will increase the block chance by 10 percent! Careful not to drop it in battle though....
add: Between The Sheets: Save the pillow talk for next shift, having this in your system while you're asleep will slowly heal you!
add: Menthol: Prevents coughing. Good for a cold or if the Chaplain gets his hands on a smoke book!
tweak: Descriptions have been edited so that if you view these under a chem master they will tell you the effect.
/:cl:

Why: Hopefully these drinks will see more use now that they have some unique effects!

----

Gives effects to a few drinks found in #37589 (Adds 5 new drinks)

### CODE REVIEW

**Comment by ExcessiveUseOfCobblestone** [comment]
The sprite was already in the game, I just had to make a new one that loops indefinitely.

**Comment by Vidiveni** [comment]
Does Alexander increase block chance by 10 percent or 10 percentage points?

**Comment by newfren** [comment]
If the shield you are carrying has a block chance of 50 it will be changed to 60, just from reading the code.

**Comment by Vidiveni** [comment]
@newfren 
I'll assume that's percentage points, for the wiki.

**Comment by Mickyan** [comment]
Not a fan at all of Kamikaze, it sounds like a funny gimmick until you realize the potential to use it as actual poison and if I'm reading this right it does a ridiculous amount of damage. Imagine someone using it in a smoke grenade on the shuttle.

**Comment by Tlaltecuhtli** [comment]
make kamikaze give you a verb that primes all the grenades on you instead and makes you shout a custom word

**Comment by Mickyan** [comment]
Just so we're on the same page I just tested it and 5u of kamikaze will put someone who's alone in a room in crit in just about 1 minute, halve that time for every person in the room.

Throwing a smoke bomb in a crowded room will make everyone drop dead as soon as the smoke dissipates (because the effect checks for line of sight)

The effect ignores internals, hardsuits, biosuits

Let's not give the bartender easy unlimited access to deadly toxins

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 1]
```diff
@@ -1562,46 +1562,95 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
 	color = "#EEF191"
 	boozepwr = 60
 	taste_description = "divine windiness"
 	glass_icon_state = "kamikaze"
 	glass_name = "Kamikaze"
-	glass_desc = "Divinely windy."
+	glass_desc = "A drink you don't exactly make friends with."
+	var/mutable_appearance/explosion_overlay
+
+/datum/reagent/consumable/ethanol/kamikaze/on_mob_add(mob/living/L)
```
Comment (Jalleo) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: I believe there is a method to make this only work if you ingest it.

**Review by Jalleo — CHANGES_REQUESTED** [review]
See what I said above.

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: errr this is only applicable on `reaction_mob` unfortunately,

I plan on making it like hearty punch where it takes a bunch to make a little.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Mickyan 

The drink was considering the user as a victim, effectively applying a flat 3 damage which is what made it so powerful as a poison. oview fixes that.

If you aren't nearby anyone while in your system, kamikaze will now give you damage as if you were a victim instead and lessen the overlay. If the overlay is completely gone, the drink metabolizes five times as fast. Every victim makes the overlay visible up to twice as much as it was initially.

**Comment by Mickyan** [comment]
@ExcessiveUseOfCobblestone I'm still not too thrilled about adding even more ways to instagib the entire shuttle but at least give it a mineral requirement (plasma/uranium) so it takes a bit of extra effort and not everyone can just hop the counter at round start or use the maint bar to mass produce it. 
It's an interesting idea but being so potentially deadly coupled with the ease of making it is a big balancing issue.

**Comment by ExcessiveUseOfCobblestone** [comment]
All you do is walk out of shuttle and either go to medical for purge chems or simply just stay away from people 3 tiles away from you and it will rapidly dissipate. Taking any form of burn DoT meds will nullify the loner damage. Remember, anyone who has this chem in their system has a blatant overlay on them that gets even more obvious the more damage that's taken. It is very obvious who to stay away from. You can also blind yourself (wirecut bedsheets or clothes to make clothes for blindfolds) to avoid the stacking damage.

Not to arms race but chem can copy a much more disrupting poison from site of choice (mostly the publicly available wiki), paste it in, and have grenades that 100% spawn with them roundstart. They don't lower their effects by walking away from people either. I don't see the big deal why bartender can't have a niche damaging chemical.

I will make the recipe like hearty punch though, I think that's a good balance for it.

**Comment by PKPenguin321** [comment]
Just give it a different effect, it should be a beer not a superpoison

**Comment by Mickyan** [comment]
I don't think walking to medbay is an option when the shuttle already left the station and you have a dozen people that have it in their system because of a smoke bomb or someone with a spray bottle

Chemistry requires specific access and is more secure than walking over a table, you also can't find chem dispensers in maintenance. Bartender drinks have always had fairly tame effects and are almost exclusively beneficial for this very reason, they were never meant to compete with chemistry

I play bartender a lot and I'm going to be the first to abuse this until it gets nerfed or removed but I'd rather skip a few steps and have something that's balanced in the first place

**Comment by ExcessiveUseOfCobblestone** [comment]
It's currently a 12input:1output ratio since one reagent takes 3 chems with a heat req, if you go out of your way to make a few units of this then kudos to you.

**Comment by AdamElTablawy** [comment]
changelings with kamikaze will rapidly become big assholes

**Comment by ExcessiveUseOfCobblestone** [comment]
Funny enough you just get close to him and they won't be able to outheal (or at least expend all their chems trying to).

**Comment by Dorsisdwarf** [comment]
so if, say, you have a shuttle with 10 people in the area of effect and they all get caught in a grenade made of this smoke, would they not take 20 damage +10 damage per second, with a lesser amount of damage being radiated to people outside the area? (shuttles are tightly packed enough that 10 people in a 7-diameter radius seems a fair amount)

**Comment by ExcessiveUseOfCobblestone** [comment]
Jesus I am stupid

It was meant to be 3 diameter not 3 radius.

**Comment by ExcessiveUseOfCobblestone** [comment]
No wonder everyone was like "oml this is op", Thanks @Dorsisdwarf

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 1 above)
Comment (Tlaltecuhtli) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: uhhh you can macro on the drink dispenser

**Comment by ExcessiveUseOfCobblestone** [comment]
mods?

**Review by SpaceManiac — CHANGES_REQUESTED** [review]
Between the Sheets and Kamikaze's `on_mob_life` need to call parent or they will never metabolize.

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 2]
```diff
@@ -1562,46 +1562,105 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+		explosion_overlay.alpha += 1
+		victimnum++
+	if(!victimnum) //No one is nearby so you take decreased damage.
+		L.adjustFireLoss(1)
+		explosion_overlay.alpha -= 1
+		if(explosion_overlay.alpha <= 0) //can be -1 until next tick
+			metabolization_rate = (initial(metabolization_rate)*5) //The overlay is gone so the drink rapidly dissapates. Congrats, you defused the -explosive- i mean drink!
+
+/datum/reagent/consumable/ethanol/kamikaze/on_mob_delete(mob/living/L)
+	L.cut_overlay(explosion_overlay)
```
Comment (SpaceManiac) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: The overlay is being removed or updating its alpha properly.

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm (hunk 2 above)
Comment (SpaceManiac) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: I tested this, the drink metabolizing does not remove the overlay, and I can't distinguish its alpha visibly changing from the default value.

Not an expert in how overlays work but you may need need to cut, edit alpha, and add again? Pester #coderbus, maybe someone knows.

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: ehhh I'll just play with it later, this pr has gone far too long.

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**